src/AutoRest.CSharp/Output/Models/Responses/ResponseHeaderGroupType.cs

```diff
@@ -21,48 +21,33 @@ internal class ResponseHeaderGroupType: TypeProvider
             "x-ms-client-request-id",
             "x-ms-request-id"
         };
+        private readonly HttpResponseHeader[] _httpResponseHeaders;
 
-        public ResponseHeaderGroupType(OperationGroup operationGroup, Operation operation, HttpResponseHeader[] httpResponseHeaders, BuildContext<DataPlaneOutputLibrary> context) : base(context)
+        public ResponseHeaderGroupType(OperationGroup operationGroup, Operation operation, HttpResponseHeader[] httpResponseHeaders, string clientName, BuildContext context) : base(context)
         {
-            ResponseHeader CreateResponseHeader(HttpResponseHeader header)
-            {
-                CSharpType type = context.TypeFactory.CreateType(header.Schema, true);
-
-                return new ResponseHeader(
-                    header.CSharpName(),
-                    header.Extensions?.HeaderCollectionPrefix ?? header.Header,
-                    type,
-                    BuilderHelpers.EscapeXmlDescription(header.Language!.Default.Description));
-            }
-
             string operationName = operation.CSharpName();
-            var clientName = context.Library.FindRestClient(operationGroup).ClientPrefix;
 
             DefaultName = clientName + operationName + "Headers";
             Description = $"Header model for {operationName}";
-            Headers = httpResponseHeaders.Select(CreateResponseHeader).ToArray();
+            _httpResponseHeaders = httpResponseHeaders;
         }
 
-        public string Description { get; }
-        public ResponseHeader[] Headers { get; }
-        protected override string DefaultName { get; }
-        protected override string DefaultAccessibility { get; } = "internal";
+        protected abstract CSharpType GetCsharpType(Schema schema, bool isNullable);
```

Comment:
isNullable not used in any of the implementations

Reply:
Also this method has the same implementation in both overrides.

Reply:
The difference is that _context has a type so we can access TypeFactory

Reply:
TypeFactory is on non-generic BuildContext.